### ISSUE TITLE
Priority queue

### ISSUE THREAD

**Issue by ancieg** [peter-wangxu/persist-queue#186]
I haven't found any other active Python projects for managing persistent queues (only little-functional or nearly archived ones).
But this has a disadvantage (important for me) - it does not have priority queues.

I would be very happy if you add support for priority queues.

**Comment by AGandhiCraniUS** [comment]
Yes Please. I really need this too. @Ancieg did you find anything?

**Comment by peter-wangxu** [comment]
I am planning to add this in next week, can you explain the use case for your scenario?

**Comment by AGandhiCraniUS** [comment]
Wow. That's great. My use case is for multiple data streams coming from a sensor that goes into python and needs to be stored (sorted by timestamp) and then retrieved for a server request. The priority queue allows me to have those streams be sorted at the time of insertion

**Comment by peter-wangxu** [comment]
@AGandhiCraniUS If you want a Timestamp based priority queue, you can try `FILOSQLiteQueue` which is in `First In Last out` order. this is very similar to PriorityQueue which use `Timestamp` as the priority key, here is the example:

```python
    def test_open_close_1000(self):
        """Write 1000 items, close, reopen checking if all items are there"""

        q = FILOSQLiteQueue(self.path, auto_commit=self.auto_commit)
        for i in range(1000):
            q.put('var%d' % i)
        self.assertEqual(1000, q.qsize())
        del q
        q = FILOSQLiteQueue(self.path)
        self.assertEqual(1000, q.qsize())
        for i in range(1000):
            data = q.get()
            self.assertEqual('var%d' % (999 - i), data)
        # assert adding another one still works
        q.put('foobar')
        data = q.get()
        self.assertEqual('foobar', data)
```

**Comment by felmoreno1726** [comment]
I've also found a need for this feature. Seems like it was actually not implemented. My requirement at the moment is using a UniqueAckQ (with persistence properties). At the moment, my queue is a JOBs queue which need to be processed by consumer workers; however, it is a good feature to have a priority queue to allow important jobs to be processed first in the queue. 


I have a workaround with two queues like:
```
    while True
        # Select queue from job_queue or priority_jobs_queue
        if priority_jobs_queue.qsize() > 0:
            selected_jobs_queue = priority_jobs_queue
        elif jobs_queue.qsize() > 0:
            selected_jobs_queue = jobs_queue
        else:
            logger.info("🔄 No jobs in the queue, waiting for new jobs")
            # Wait for a job to be available
            sleep(5)
            continue

        job = selected_jobs_queue.get()

        # <dispatch job>
```

However, I don't find this solution elegant as a single queue with priority mechanisms. The continuous checking for a new item with .qsize() is busy waiting, and the qsize() calls are also prone to race conditions.

**Comment by peter-wangxu** [comment]
> I've also found a need for this feature. Seems like it was actually not implemented. My requirement at the moment is using a UniqueAckQ (with persistence properties). At the moment, my queue is a JOBs queue which need to be processed by consumer workers; however, it is a good feature to have a priority queue to allow important jobs to be processed first in the queue.
> 
> I have a workaround with two queues like:
> 
> ```
>     while True
>         # Select queue from job_queue or priority_jobs_queue
>         if priority_jobs_queue.qsize() > 0:
>             selected_jobs_queue = priority_jobs_queue
>         elif jobs_queue.qsize() > 0:
>             selected_jobs_queue = jobs_queue
>         else:
>             logger.info("🔄 No jobs in the queue, waiting for new jobs")
>             # Wait for a job to be available
>             sleep(5)
>             continue
> 
>         job = selected_jobs_queue.get()
> 
>         # <dispatch job>
> ```
> 
> However, I don't find this solution elegant as a single queue with priority mechanisms. The continuous checking for a new item with .qsize() is busy waiting, and the qsize() calls are also prone to race conditions.

I added PrioprityQueue as requested, let me know if fit your demand

**Comment by felmoreno1726** [comment]
Is there a way to get the features from UniqueAwqQueue and PriorityQueue? Would it work to make an object that inherits from both?
